loader/cJSON.c

```diff
@@ -428,32 +423,35 @@ static char *print_string_ptr(const char *str, printbuffer *p) {
     return out;
 }
 /* Invote print_string_ptr (which is useful) on an item. */
-static char *print_string(cJSON *item, printbuffer *p) { return print_string_ptr(item->valuestring, p); }
+char *print_string(const struct loader_instance *instance, cJSON *item, printbuffer *p) {
+    return print_string_ptr(instance, item->valuestring, p);
+}
 
 /* Predeclare these prototypes. */
-static const char *parse_value(cJSON *item, const char *value);
-static char *print_value(cJSON *item, int depth, int fmt, printbuffer *p);
-static const char *parse_array(cJSON *item, const char *value);
-static char *print_array(cJSON *item, int depth, int fmt, printbuffer *p);
-static const char *parse_object(cJSON *item, const char *value);
-static char *print_object(cJSON *item, int depth, int fmt, printbuffer *p);
+const char *parse_value(const struct loader_instance *instance, cJSON *item, const char *value);
+char *print_value(const struct loader_instance *instance, cJSON *item, int depth, int fmt, printbuffer *p);
+const char *parse_array(const struct loader_instance *instance, cJSON *item, const char *value);
+char *print_array(const struct loader_instance *instance, cJSON *item, int depth, int fmt, printbuffer *p);
+const char *parse_object(const struct loader_instance *instance, cJSON *item, const char *value);
+char *print_object(const struct loader_instance *instance, cJSON *item, int depth, int fmt, printbuffer *p);
```

Comment:
Why'd you remove all the static modifiers?

Reply:
The purpose static serves it to make sure other translation units aren't able to call the functions. Since these functions are not defined in the header file, it doesn't serve much of a purpose.